packages/lodestar/src/util/validation/attestation.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  EpochContext,
+  getAttestingIndices,
+  getCurrentSlot,
+  getIndexedAttestation,
+  isValidIndexedAttestation
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {ATTESTATION_PROPAGATION_SLOT_RANGE} from "../../constants";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {Attestation, BeaconState} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db/api";
+import {assert} from "@chainsafe/lodestar-utils";
+import {processSlots} from "@chainsafe/lodestar-beacon-state-transition/lib/fast/slot";
+
+/**
+ * is ready to be included in block
+ */
+export function hasValidAttestationSlot(config: IBeaconConfig, genesisTime: number, attestation: Attestation): boolean {
+  const currentSlot = getCurrentSlot(config, genesisTime);
+  return attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= currentSlot
+        && currentSlot >= attestation.data.slot;
+}
+
+export function isUnaggregatedAttestation(
+  config: IBeaconConfig, state: BeaconState, epochCtx: EpochContext, attestation: Attestation
+): boolean {
+  if (state.slot < attestation.data.slot) {
+    processSlots(epochCtx, state, attestation.data.slot);
+  }
+  // Make sure this is unaggregated attestation
+  return getAttestingIndices(config, state, attestation.data, attestation.aggregationBits).length === 1;
```

Comment:
`getAttestingIndices` should be split in two functions, one looks like:
`getAttestingIndicesFromCommittee(committee: ValidatorIndex[], bits: BitList): ValidatorIndex`
So we can use `epochCtx.getBeaconCommittee` instead of slow committee calculation here.